docs/src/app/(public)/(content)/react/handbook/composition/page.mdx

```diff
@@ -0,0 +1,79 @@
+# Composition
+
+<Subtitle>A guide to composing Base UI components with your own React components.</Subtitle>
+<Meta
+  name="description"
+  content="A guide to composing Base UI components with your own React components."
+/>
+
+### Composing custom React components
+
+The `render` prop lets you override a component's rendered element with your own React component. For example, `Menu.Trigger` renders a `<button>` by default. The code snippet below shows how to use a custom component instead.
+
+```tsx title="index.tsx"
+<Menu.Trigger render={<MyFancyButton size="md" />} />
+```
+
+The custom component must forward a `ref`, and spread all the received props, to its underlying DOM node.
+
+### Composing multiple components
+
+In situations where you need to compose multiple Base UI components with custom React components, `render` props can be nested as deeply as necessary. Working with Tooltip is a common example.
+
+```tsx title="index.tsx"
+<Dialog.Root>
+  <Tooltip.Root>
+    <Tooltip.Trigger
+    {/* prettier-ignore */}
+      render={
+        <Dialog.Trigger
+          render={
+            <MyFancyButton size="md" />
+          }
+        />
+      }
+    />
+    <Tooltip.Portal>...</Tooltip.Portal>
+  </Tooltip.Root>
+  <Dialog.Portal>...</Dialog.Portal>
+</Dialog.Root>
+```
+
+### Changing the default rendered element
+
+You can also use the `render` prop to override the rendered element of the component. For example, `Menu.Item` renders a `<div>` by default.
+The code snippet below shows how to render an `<a>` instead.
+
+```tsx title="index.tsx"
+import * as React from 'react';
+import { Menu } from '@base-ui-components/react/menu';
+
+export default () => (
+  <Menu.Root>
+    <Menu.Trigger>Song</Menu.Trigger>
+    <Menu.Portal>
+      <Menu.Positioner>
+        <Menu.Popup>
+          <Menu.Item render={<a href="base-ui.com" />}>Add to Library</Menu.Item>
+        </Menu.Popup>
+      </Menu.Positioner>
+    </Menu.Portal>
+  </Menu.Root>
+);
+```
+
+Changing the default rendered element can break important accessibility features like focus management, screen reader announcements, and keyboard events. Each Base UI component renders the most appropriate element by default, and in most cases, rendering a different element is not recommended.
```

Comment:
I don't think we need to scare people away from the render prop here, since most components implement the right semantics using ARIA roles and keyboard accessibility doesn't hinge on the tag either. Composing with custom react components is much more likely to break something

```suggestion
Each Base UI component renders the most appropriate element by default, and in most cases, rendering a different element is recommended only on a case-by-case basis.
```

Reply:
I agree with the original suggestion more. I don't think we should encourage anyone to change the rendered element. We can save ourselves some trouble when the issues that something exotic doesn't work start coming.